hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/internal/DataSourceInternalWriterHelper.java

```diff
@@ -85,14 +97,49 @@ public void commit(List<WriteStatus> writeStatuses) {
     try {
       List<HoodieWriteStat> writeStatList = writeStatuses.stream().map(WriteStatus::getStat).collect(Collectors.toList());
       writeClient.commitStats(instantTime, writeStatList, Option.of(extraMetadata),
-          CommitUtils.getCommitActionType(operationType, metaClient.getTableType()));
+          CommitUtils.getCommitActionType(operationType, metaClient.getTableType()), getReplacedFileIds(writeStatuses), Option.empty());
     } catch (Exception ioe) {
       throw new HoodieException(ioe.getMessage(), ioe);
     } finally {
       writeClient.close();
     }
   }
 
+  private Map<String, List<String>> getReplacedFileIds(List<WriteStatus> writeStatuses) {
```

Comment:
Sorry, I still think keeping `getReplacedFileIds` in each executor is more cleaner and reasonable.

Reply:
> Sorry, I still think keeping `getReplacedFileIds` in each executor is more cleaner and reasonable.

Your suggestion is reasonable. However, since the current executor is only used to do the bulk_insert pre-preparation, and then the rest of the writes and commits are handed over to `DataSourceInternalWriterHelper`, `executor::getReplaceFileIds` are not called. Because executor is not responsible for committing, the `getReplaceFileId` method is not required. My original intention was just to converge the actual execution and commit code paths of `bulk_insert`/`insert_overwrite`/`insert_overwrite_table` together, instead of this: Some use `HoodieDatasetBulkInsertHelper`, some `DataSourceInternalWriterHelper`. 
So do you have any good suggestions? Looking forward to your reply~

Reply:
>  is only used to do the bulk_insert pre-preparation, and then the rest of the writes and commits are handed over to DataSourceInternalWriterHelper

Can we move the writes and commits inside the executors?

Reply:
> Can we move the writes and commits inside the executors?

Of course we can, then we don't need `DataSourceInternalWriterHelper ` to perform writes and commits, all writes and commits are consolidated inside executor, and I'll make changes to follow that logic

Reply:
+1 about `Can we move the writes and commits inside the executors?` It is better to let different executor to take care of their own replace logic instead of union together in a big switch or if-else